### ISSUE TITLE
[tooling]: NodeJS scripts to make exegen.js executable

### ISSUE THREAD

**Issue by brakmic** [microsoft/BosqueLanguage#251]
**What feature do you think could be better?**

I'd recommend a simple NodeJS script that makes *exegen.js* executable.

**Why do you think the current design is unsatisfactory? Please provide a code example if possible.**

Due to regular recompiling of the *exegen.js* script, I have to adapt it again and again (*chmod +x*, and also the shebang entry *#!/usr/bin/env node* in its header).

**Do you have a proposed alternative design?**

I have a [branch](https://github.com/brakmic/BosqueLanguage/commit/08e4c65050eec3d196de69f6dfb4005a94ff03f8) that contains following changes:

* *impl/scripts/make-exe.js* script that appends `shebang` and `chmod`s file flags.
* two new packages added to `package.json`
* a new runscript added to `package.json`

After the initial `npm run build` execute `npm run make-exe` to run the additonal script.

Now you can compile with just `exegen.js -c clang++ -o a.out a.bsq`

**Comment by mrkmarron** [comment]
This seems useful but I would like to make it more comprehensive and set it up to address some future tasks as well. In particular can we move all the current scripts into a "scripts" known directory and then just have "thin" bash or powershell scripts that call them nicely. With this we can add a directory to the thin scripts on the path and include an "install" script to build, setup the vscode extension as needed, and setup the path.

In the future we want to add support to build N-API Node.js modules from Bosque so having a script that does an auto-setup of a skeleton directory/project for this would be another one to have here.

**Comment by brakmic** [comment]
Hi @mrkmarron ,

Finally got some time for Bosque! 

I've opened a PR that would add a new npm script that can do platform-independent "installations". Actually, it's just a symlink via `npm link` and a little script to set execution flags for macOS/Linux (Windows would simply ignore it).

Not sure if this solution goes in the correct direction.

**Comment by mrkmarron** [comment]
running `npm run-script make-exe` can trigger a warning that `chmod` has an unpatched vulnerability.

```
 === npm audit security report ===

  Low             Prototype Pollution
  Package         deep-extend
  Patched in      >=0.5.1
  Dependency of   chmod [dev]
  Path            chmod > deep-extend
  More info       https://npmjs.com/advisories/612
```
It looks like chmod is abandoned. Is there an alternative version or can we just vendor this into the scripts in the `/build` directory?

**Comment by brakmic** [comment]
> running `npm run-script make-exe` can trigger a warning that `chmod` has an unpatched vulnerability.
> 
> ```
>  === npm audit security report ===
> 
>   Low             Prototype Pollution
>   Package         deep-extend
>   Patched in      >=0.5.1
>   Dependency of   chmod [dev]
>   Path            chmod > deep-extend
>   More info       https://npmjs.com/advisories/612
> ```
> 
> It looks like chmod is abandoned. Is there an alternative version or can we just vendor this into the scripts in the `/build` directory?

I will look into it.

**Comment by brakmic** [comment]
@mrkmarron 

The `chmod` package is irrelevant anyway as Unix-like OSes provide their own methods to take care of file execution flags.

Have opened a new PR.
